weights.py

```diff
@@ -189,19 +220,45 @@ def download_safetensors(url: str, path: Path):
         raise RuntimeError(f"Failed to download safetensors file: {e}")
 
 
-def make_download_url(url: str) -> str:
+def make_download_url(
+    url: str, hf_api_token: str | None = None, civitai_api_token: str | None = None
+) -> str:
     if url.startswith("data:"):
         return url
-    if m := re.match(r"^(?:https?://)?huggingface\.co/([^/]+)/([^/]+)/?$", url):
-        owner, model_name = m.groups()
-        return make_huggingface_download_url(owner, model_name)
+    if m := re.match(
+        r"^(?:https?://)?huggingface\.co/([^/]+)/([^/]+)(?:/([^/]+\.safetensors))?/?$",
```

Comment:
we shouldn't make a breaking change here; what we can do instead is use the HfApi to list the files in a repo and then do the same check for the existence of only one `.safetensors` file. 

should look something like this: 

```
from huggingface_hub import HfApi

api = HfApi()
files = api.list_repo_files(repo_id)
sft_files = [file for file in files if '.safetensors' in file]
if len(sft_files) == 1:
    hf_hub_download(repo_id=repo_id, filename=sft_files[0])
else:
   # throw errors if there are 0 files or > 1 files
   ```